commander3/src/comm_tod_mod.f90

```diff
@@ -569,46 +710,55 @@ subroutine read_hdf_scan(self, tod, filename, scan, ndet, detlabels)
     ! Read common scan data
     call read_hdf(file, slabel // "/common/vsun",  self%v_sun)
     call read_hdf(file, slabel // "/common/time",  self%t0)
-    call read_hdf(file, slabel // "/common/satpos",  self%satpos)
+    call read_hdf(file, slabel // "/common/satpos",  self%satpos, opt=.true.)
     call wall_time(t2)
     t_tot(1) = t2-t1
 
+    !write(*,*) self%t0(1), real(self%v_sun), "# v"
+
     ! Read detector scans
     allocate(self%d(ndet), buffer_sp(n))
     do i = 1, ndet
+       allocate(self%d(i)%psi(nhorn), self%d(i)%pix(nhorn))
+
        call wall_time(t1)
        field = detlabels(i)
        call wall_time(t2)
        t_tot(2) = t_tot(2) + t2-t1
        call wall_time(t1)
-       allocate(self%d(i)%tod(m))
        self%d(i)%label = trim(field)
        call read_hdf(file, slabel // "/" // trim(field) // "/scalars",   scalars)
        self%d(i)%gain_def   = scalars(1)
-       self%d(i)%sigma0_def = scalars(2) * self%d(i)%gain_def  ! To get sigma0 in uncalibrated units
+       self%d(i)%sigma0_def = scalars(2) * abs(self%d(i)%gain_def)  ! To get sigma0 in uncalibrated units
        self%d(i)%fknee_def  = scalars(3)
        self%d(i)%alpha_def  = scalars(4)
        self%d(i)%gain       = self%d(i)%gain_def
        self%d(i)%sigma0     = self%d(i)%sigma0_def
        self%d(i)%fknee      = self%d(i)%fknee_def
        self%d(i)%alpha      = self%d(i)%alpha_def
+
        call wall_time(t2)
        t_tot(3) = t_tot(3) + t2-t1
        call wall_time(t1)
-       call read_hdf(file, slabel // "/" // trim(field) // "/tod",    buffer_sp)
-       if (tod%halfring_split == 2 )then
-         self%d(i)%tod = buffer_sp(m+1:2*m)
-       else
-         self%d(i)%tod = buffer_sp(1:m)
-       end if
+       !call read_hdf(file, slabel // "/" // trim(field) // "/tod",    buffer_sp)
+       !allocate(self%d(i)%tod(m))
+       !if (tod%halfring_split == 2 )then
+       !  self%d(i)%tod = buffer_sp(m+1:2*m)
+       !else
+       !  self%d(i)%tod = buffer_sp(1:m)
+       !end if
        call wall_time(t2)
        t_tot(4) = t_tot(4) + t2-t1
 
+
        ! Read Huffman coded data arrays
        call wall_time(t1)
-       call read_hdf_opaque(file, slabel // "/" // trim(field) // "/pix",  self%d(i)%pix)
-       call read_hdf_opaque(file, slabel // "/" // trim(field) // "/psi",  self%d(i)%psi)
+       do j = 1, nhorn
+         call read_hdf_opaque(file, slabel // "/" // trim(field) // "/pix" // char(j+64),  self%d(i)%pix(j)%p)
+         call read_hdf_opaque(file, slabel // "/" // trim(field) // "/psi" // char(j+64),  self%d(i)%psi(j)%p)
+       end do
        call read_hdf_opaque(file, slabel // "/" // trim(field) // "/flag", self%d(i)%flag)
+       call read_hdf_opaque(file, slabel // "/" // trim(field) // "/tod", self%d(i)%tod)
```

Comment:
This assumes the TOD is compressed. We need to support both options

Reply:
I think @haraltho did this, I can try and copy his approach